### PR TITLE
Feature/send put request from consumer to api

### PR DESCRIPTION
**Pull request by KohareCH**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/psc/delta/service/api/ApiClientServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * Service that sends REST requests via private SDK.
+ */
+@Primary
+@Service
+public class ApiClientServiceImpl extends BaseApiClientServiceImpl implements ApiClientService {
+
+    @Value("${api.pscFullRecord-data-api-key}")
+    private String chsApiKey;
+
+    @Value("${api.api-url}")
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/psc/delta/service/api/ApiClientServiceImpl.java:29]: Have you added these configs to the application.yml file?

File: src/main/java/uk/gov/companieshouse/psc/delta/service/api/ApiClientServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+
+/**
+ * Service that sends REST requests via private SDK.
+ */
+@Primary
+@Service
+public class ApiClientServiceImpl extends BaseApiClientServiceImpl implements ApiClientService {
+
+    @Value("${api.pscFullRecord-data-api-key}")
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/psc/delta/service/api/ApiClientServiceImpl.java:26]: I think you just need this to be `api.psc-data-api-key` the consumer will only ever use one key

File: src/main/java/uk/gov/companieshouse/psc/delta/service/api/BaseApiClientServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        "400 BAD_REQUEST response received from psc-data-api";
+                logger.errorContext(logContext, msg, ex, logMap);
+                throw new NonRetryableErrorException(msg, ex);
+            }
+//            else if (ex.getStatusCode() == HttpStatus.NOT_FOUND.value()
+//                    && operationName.equals("deleteDisqualification")) {
+//                String msg =
+//                        "404 NOT_FOUND response received from psc-data-api";
+//                throw new RetryableErrorException(msg);
+//            }
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/psc/delta/service/api/BaseApiClientServiceImpl.java:64]: Is there a reason this is commented out?

File: src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Process PSC Statement Delta message.
+     */
+    public void processDelta(Message<PscDelta> pscDelta) {
+        final MessageHeaders headers = pscDelta.getHeaders();
+        final PscDelta payload = pscDelta.getPayload();
+        logger.info(format("Successfully extracted PSC Delta of %s", payload.toString()));
+
+//        apiClientService.putPscFullRecord(contextId, companyId, notficationId, fullRecordCompanyPSCApi);
```
Comment (amartin7663) [inline src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java:34]: Is there a reason this is commented out?

Comment (KohareCH) [reply src/main/java/uk/gov/companieshouse/psc/delta/processor/PscDeltaProcessor.java]: Currently left for Jp, for his changes for the processor, left commented out

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**